### PR TITLE
DCMAW 9420 Improve sonar test coverage

### PR DESCRIPTION
**Pull request by ThomasIent**
**Ticket Link:** [DCMAW 9420](https://govukverify.atlassian.net/browse/DCMAW-9420)
**Description Of Changes:** 

I have focused on getting the test coverage up, and not on making the code good. I didn't want to change the public facing API, so there are a lot of small pure static functions in order to test paths, ideally we would refactor some pre-existing classes to better encapsulate these (both Util files shouldn't have to exist).
There is also some small parts that aren't being used (in this project) and can possibly be deleted, but we would need to check the other projects as they are public.

**Evidence:**

- [DCMAW-9420-JacocoConnectedReport.pdf](https://github.com/user-attachments/files/16725829/DCMAW-9420-JacocoConnectedReport.pdf)

**Documentation:** No Documentation
**Definition Of Done Checklist:**
- [ ] Unit tests written to at least 80% code coverage
- [ ] Relevant integration and end-to-end tests are written
- [x] Tests cover each acceptance criteria on the user story
- [ ] Demo completed by running code locally, demoing to a code reviewer & designer
- [ ] Sonar coverage gates met 
- [x] Feature flag is enabled in staging and manually tested

### CODE REVIEW

File: app/src/androidTest/java/uk/gov/android/authentication/AppAuthSessionTest.kt [hunk 1]
```diff
@@ -1,102 +1,68 @@
...
+        // Then create the AuthorisationIntent and launch
+        verify(launcher).launch(any())
     }
 
-    @Ignore("Will be fixed in refactor")
-    @Test
-    fun unsuccessfulFinaliseCallbackResponse() {
-        val error = assertThrows(AuthenticationError::class.java) {
-            loginSession.finalise(Intent()) {}
+    @Test(expected = IllegalArgumentException::class)
```
Comment (alex-bradbury) [inline app/src/androidTest/java/uk/gov/android/authentication/AppAuthSessionTest.kt:45]: This is cool!

File: app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt [hunk 2]
```diff
@@ -20,85 +15,27 @@ class AppAuthSession(
...
-            accessTokenExpirationTime =
-            requireNotNull(response.accessTokenExpirationTime) {
-                "Token expiry must not be empty"
-            },
-            idToken = response.idToken,
-            refreshToken = response.refreshToken
-        )
+    companion object {
+        @Suppress("unused")
+        const val REQUEST_CODE_AUTH = 418
```
Comment (alex-bradbury) [inline app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt:39]: Don't think we need this anymore... Think it was a remontant from using `Activity.launchActivityForResult()`

File: app/src/androidTest/java/uk/gov/android/authentication/AppAuthSessionTest.kt (hunk 1 above)
Comment (ThomasIent) [reply app/src/androidTest/java/uk/gov/android/authentication/AppAuthSessionTest.kt:45]: It's nice if you don't need to check any exception internals, but they've removed it in JUnit5.

File: app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt (hunk 2 above)
Comment (ThomasIent) [reply app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt]: Removed it

**Review by alex-bradbury — APPROVED** [review]
Looks great, really nice work 🚀

**Pull request merged**